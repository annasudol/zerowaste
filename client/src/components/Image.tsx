import * as React from "react";
import cx from 'classnames';

export interface ImageProps {
    alt: string;
    size: 'small' | 'medium' | 'large' | 'full';
    src: string;
    className?: string;
}

export const Image: React.FunctionComponent<ImageProps> = ({ alt, size, src, className }): React.ReactElement => {
    const sizeOptions = size === 'small' ? 'w-16' : size === 'medium' ? 'w-40' : 'w-64';
    return <img src={src} alt={alt} className={cx(sizeOptions, className)} />;
};
