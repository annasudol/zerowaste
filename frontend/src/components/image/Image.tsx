import React, { FC, ReactElement } from 'react';
import cx from 'classnames';

export interface ImageProps {
    alt: string;
    size?: 'small' | 'medium' | 'large' | 'full';
    src: string;
    className?: string;
}

export const Image: FC <ImageProps> = ({alt, size = 'medium', src, className}): ReactElement => {
    const sizeOptions = size === 'small' ? 'w-32' : size === 'medium' ? 'w-64' : '300';
    return <img src={src} alt={alt} className={cx(sizeOptions, className)} />;
};
