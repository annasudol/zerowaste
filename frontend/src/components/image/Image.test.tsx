import { mount } from 'enzyme';
import React from 'react';

import toJson from 'enzyme-to-json';
import { Image } from './Image';

describe('component', (): void => {
  describe('Image', (): void => {
    it('match snapshot with medium size image', (): void => {
      const image = mount(<Image src={require('url')} size="medium" alt="test" />);
      expect(toJson(image)).toMatchSnapshot();
    });
    it('match snapshot with small size image', (): void => {
      const image = mount(<Image src={require('url')} size="small" alt="test" />);
      expect(toJson(image)).toMatchSnapshot();
    });
    it('match snapshot with large size and rounded image', (): void => {
      const image = mount(<Image src={require('url')} size="large" alt="test"/>);
      expect(toJson(image)).toMatchSnapshot();
    });
  });
});