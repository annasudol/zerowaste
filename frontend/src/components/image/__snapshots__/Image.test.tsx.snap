// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Image match snapshot with large size and rounded image 1`] = `
<Image
  alt="test"
  size="large"
  src={
    Object {
      "URL": [Function],
      "URLSearchParams": [Function],
      "Url": [Function],
      "domainToASCII": [Function],
      "domainToUnicode": [Function],
      "fileURLToPath": [Function],
      "format": [Function],
      "parse": [Function],
      "pathToFileURL": [Function],
      "resolve": [Function],
      "resolveObject": [Function],
    }
  }
>
  <img
    alt="test"
    className="300"
    src={
      Object {
        "URL": [Function],
        "URLSearchParams": [Function],
        "Url": [Function],
        "domainToASCII": [Function],
        "domainToUnicode": [Function],
        "fileURLToPath": [Function],
        "format": [Function],
        "parse": [Function],
        "pathToFileURL": [Function],
        "resolve": [Function],
        "resolveObject": [Function],
      }
    }
  />
</Image>
`;

exports[`component Image match snapshot with medium size image 1`] = `
<Image
  alt="test"
  size="medium"
  src={
    Object {
      "URL": [Function],
      "URLSearchParams": [Function],
      "Url": [Function],
      "domainToASCII": [Function],
      "domainToUnicode": [Function],
      "fileURLToPath": [Function],
      "format": [Function],
      "parse": [Function],
      "pathToFileURL": [Function],
      "resolve": [Function],
      "resolveObject": [Function],
    }
  }
>
  <img
    alt="test"
    className="w-64"
    src={
      Object {
        "URL": [Function],
        "URLSearchParams": [Function],
        "Url": [Function],
        "domainToASCII": [Function],
        "domainToUnicode": [Function],
        "fileURLToPath": [Function],
        "format": [Function],
        "parse": [Function],
        "pathToFileURL": [Function],
        "resolve": [Function],
        "resolveObject": [Function],
      }
    }
  />
</Image>
`;

exports[`component Image match snapshot with small size image 1`] = `
<Image
  alt="test"
  size="small"
  src={
    Object {
      "URL": [Function],
      "URLSearchParams": [Function],
      "Url": [Function],
      "domainToASCII": [Function],
      "domainToUnicode": [Function],
      "fileURLToPath": [Function],
      "format": [Function],
      "parse": [Function],
      "pathToFileURL": [Function],
      "resolve": [Function],
      "resolveObject": [Function],
    }
  }
>
  <img
    alt="test"
    className="w-32"
    src={
      Object {
        "URL": [Function],
        "URLSearchParams": [Function],
        "Url": [Function],
        "domainToASCII": [Function],
        "domainToUnicode": [Function],
        "fileURLToPath": [Function],
        "format": [Function],
        "parse": [Function],
        "pathToFileURL": [Function],
        "resolve": [Function],
        "resolveObject": [Function],
      }
    }
  />
</Image>
`;
